pennylane/tape/operation_recorder.py

```diff
@@ -63,20 +72,32 @@ def _process_queue(self):
         self.obs = new_tape.observables
 
     def __str__(self):
-        output = ""
-        output += "Operations\n"
-        output += "==========\n"
-        for op in self.ops:
-            output += repr(op) + "\n"
-
-        output += "\n"
-        output += "Observables\n"
-        output += "===========\n"
-        for op in self.obs:
-            output += repr(op) + "\n"
-
-        return output
+        return "\n".join(
+            [
+                "Operations",
+                "==========",
+                *[repr(op) for op in self.ops],
+                "",
+                "Observables",
+                "===========",
+                *[repr(op) for op in self.obs],
+                "",
+            ]
+        )
 
     @property
     def queue(self):
         return self.ops + self.obs
+
+    def __getitem__(self, key):
+        """
+        Overrides the default because QuantumTape is both a QuantumScript and an AnnotatedQueue.
+        If key is an int, the caller is likely indexing the backing QuantumScript. Otherwise, the
+        caller is likely indexing the backing AnnotatedQueue.
+        """
+        if isinstance(key, int):
+            return QuantumScript.__getitem__(self, key)

```

Comment:
no one ever actually did `recorder[idx]` in tests, so maybe it's only intended to be indexed as an `AnnotatedQueue` (aka via object key, and not via list position). I figured I might as well copy this over, but I had to add a line to a test in `test_operation_recorder.py` to make sure this `QuantumScript.__getitem__()` line is ever hit by tests.